include/zephyr/drivers/cellular.h

```diff
@@ -90,6 +121,36 @@ int cellular_configure_networks(const struct device *dev, const struct cellular_
 int cellular_get_supported_networks(const struct device *dev,
 				    const struct cellular_network **networks, uint8_t *size);
 
+/**
+ * @brief Get signal for the device
+ *
+ * @param dev Cellular network device instance
+ * @param type Type of the signal information requested
+ * @param data Pointer to one of cellular_signal_{rssi,percentage,quality}
+ *
+ * @retval 0 if successful.
+ * @retval -ENOTSUP if API is not supported by cellular network device.
+ * @retval -ENODATA if device is not in a state where signal can be polled
+ * @retval Negative errno-code otherwise.
+ */
+int cellular_get_signal(const struct device *dev, const enum cellular_signal_type type, void *data);
```

Comment:
If we can agree to use an integer here, we should use `int16_t *value`, in place of `void *data`, its quite a bit safer than void* if we don't need it :)

Reply:
sure